### DISCUSSION TITLE
关于序列推荐数据预处理问题

### DISCUSSION THREAD

**Discussion by kuzma-long** [RUCAIBox/RecBole#1291] [Q&A] [answered]
Recbole对每一个输入序列做的预处理是，不足最大序列长度的在右边补0，但一般论文里都是在左边补0，这个顺序颠倒对结果不会有影响吗？

**Comment by hyp1231** [comment, marked as answer]
一方面，我们取最后一个时刻的输出时，使用了 `gather_index` 这个函数，保证取到的不是 0 处的 hidden state；另一方面，我个人的实验结果显示，position encoding 在对左/右补零的序列，尽管计算上不同，但是结果没有显著的差异。
